dev-tools/docker/linux_aarch64_cross_image/Dockerfile

```diff
@@ -30,8 +33,8 @@ RUN \
   ln -s usr/lib64 lib64
 
 # For compiling with hardening and optimisation
-ENV CFLAGS "-g -O3 -fstack-protector -D_FORTIFY_SOURCE=2 -msse4.2 -mfpmath=sse"
-ENV CXXFLAGS "-g -O3 -fstack-protector -D_FORTIFY_SOURCE=2 -msse4.2 -mfpmath=sse"
+ENV CFLAGS "-g -O3 -fstack-protector -D_FORTIFY_SOURCE=2 -march=armv8-a+crc+crypto"
+ENV CXXFLAGS "-g -O3 -fstack-protector -D_FORTIFY_SOURCE=2 -march=armv8-a+crc+crypto"
```

Comment:
I think this is wrong.  In this image we're building tools that will run on `x86_64` in order to subsequently do cross compilation to `aarch64`.  So for the build of those tools that will run on `x86_64` we need options that will make the tools run fast on `x86_64`.

Arguably the options here for building `gcc` and `binutils` should be `CFLAGS_FOR_BUILD` and `CXXFLAGS_FOR_BUILD`.  It seems that `configure.sh` will copy `CFLAGS` to both `CFLAGS_FOR_BUILD` _and_ also `CFLAGS_FOR_TARGET` and `CXXFLAGS_FOR_TARGET` if those more specialised variables aren't specified.  But then `CFLAGS_FOR_TARGET` and `CXXFLAGS_FOR_TARGET` don't seem to actually affect the default options that the resulting cross compiler will apply when building for the target (or at least not the ones we're specifying here).

Arguably it would be more correct to specify `CFLAGS_FOR_BUILD` and `CXXFLAGS_FOR_BUILD` here, then replace them with `CFLAGS` and `CXXFLAGS` when building `patchelf` lower down (as that doesn't know about `CFLAGS_FOR_BUILD` and `CXXFLAGS_FOR_BUILD`).  But as far as I can see the way it's being done at the moment doesn't actually cause a problem.  We specify `-march=armv8-a+crc+crypto` in the compiler options when we use it to build our code for `aarch64`.

So I would just put this back how it was.